jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -143,9 +208,15 @@ def _jwst_to_glue_data(file_obj, ext, data_label):
         if ext in ('sci', 'asdf'):
             ext = 'data'
 
+    data, new_data_label = _jwst2data(data_label, ext, file_obj)
+
+    yield data, new_data_label
+
+
+def _jwst2data(data_label, ext, file_obj):
```

Comment:
The `return` function is common to both the 1-extension and the `*` all-extension cases. It contains the relevant machinery. The two separate cases are necessary because the `yeld` must happen within the loop over extensions.

Reply:
Made all these calls consistent with each other.